### PR TITLE
Feature: Event Visibility

### LINKED ISSUE #338: Feature: Allow for Unpublishing Imported Organizations
This came from the Slack thread about some of our [HG orgs](https://hackgreenville.com/orgs) posting a mix of commercial and non-commercial events.  

We could allow for unpublishing an imported event with a new field on the events.

Or, as a more advanced feature, we could tag certain organizations so all of their imported events are automatically unpublished and requiring manual review.

Flywheel was the example that sparked the conversation, since they have a mix of free meetup-style Flywheel Cafe events and some more commercial open-house events for the coworking side posted to the same Eventbrite account.

This is related to https://github.com/hackgvl/hackgreenville-com/issues/240 and https://github.com/hackgvl/hackgreenville-com/pull/330 and this feature would give us a good deal of ability to manually work around. Though, we've asked the HG Committee to talk about how we might deal with some of the grey area of hybrid organizations who might be posting non-commercial and commercial events to the same events account that we're automatically importing.  

Here's the Slack conversation
**Eric:**
image

### PR DESCRIPTION
**Pull request by bogdankharchenko**
Resolves #338

### CODE REVIEW

File: resources/views/livewire/hackgreenville-timeline.blade.php [hunk 1]
```diff
@@ -27,14 +27,14 @@
...
-                                    <i class="fa fa-calendar"  ></i> {{ \Carbon\Carbon::parse($event['active_at'])->format('M/d h:i A') }}
+                                    <i class="fa fa-calendar"></i> {{ \Carbon\Carbon::parse($event['active_at'])->format('M/d h:i A') }}
                                 </small>
                             </p>
                         </div>
                         <div class="timeline-body">
                             <div>
                                 <button
-                                    onClick="showMoreTimeline('{{addslashes($event->toJson()) }}')"
+                                    onClick="showMoreTimeline(@js(['title' => $event->event_name, 'html' => $event->description]))"
```
Comment (allella) [inline resources/views/livewire/hackgreenville-timeline.blade.php:37]: Is this output being sanitized?  The previous command had an addslashes and I wanted to check that we have some XSS protection on the HTML coming from the orgs.

File: resources/views/livewire/hackgreenville-timeline.blade.php [hunk 2]
```diff
@@ -12,13 +12,13 @@
             @foreach ($events as $event)
                 <li class="timeline-inverted">
                     <div class="timeline-badge bg-success">
-                        <i class="fa fa-calendar" ></i>
+                        <i class="fa fa-calendar"></i>
```
Comment (allella) [inline resources/views/livewire/hackgreenville-timeline.blade.php:15]: This Blade file has Livewire in the path.  It looked like we were removing Livewire in another PR.  Is it worth renaming this view?

resources/views/livewire/hackgreenville-timeline.blade.php

File: resources/views/livewire/hackgreenville-timeline.blade.php (hunk 1 above)
Comment (bogdankharchenko) [reply resources/views/livewire/hackgreenville-timeline.blade.php:37]: Yes it is.

File: resources/views/livewire/hackgreenville-timeline.blade.php (hunk 2 above)
Comment (bogdankharchenko) [reply resources/views/livewire/hackgreenville-timeline.blade.php:15]: I will have to resolve merge conflicts anyway - it depends what we merge in first, either way I will have to tweak it.

**Comment by allella** [comment]
@bogdankharchenko preference on merging this or the Livewire PR first?

**Comment by bogdankharchenko** [comment]
Let's do livewire one first 👍

File: app/Http/Controllers/Api/CalendarController.php [hunk 3]
```diff
@@ -23,6 +23,7 @@ public function index(CalendarContract $calendar)
         $end   = Carbon::parse(request('end'));
 
         Event::datesBetween($start, $end)
+            ->with('organization', 'venue.state')
```
Comment (allella) [inline app/Http/Controllers/Api/CalendarController.php:26]: Is this requiring a venue state to be shown?  If so, how does that impact virtual events?

Comment (allella) [reply app/Http/Controllers/Api/CalendarController.php:26]: @bogdankharchenko

Comment (bogdankharchenko) [reply app/Http/Controllers/Api/CalendarController.php:26]: It is not requiring -- its just trying to load it if exists.

**Review by allella — APPROVED** [review]

**Pull request merged**